### DISCUSSION TITLE
pbrMetallicRoughness is undefined

### DISCUSSION THREAD

**Discussion by shamsail** [google/model-viewer#2660] [Q&A]
Hi, I have some models that I was seeing on modelviewer/editor. Mostly models are being displayed properly and material/texture tab is working fine (I can see the taxtures and materials and all from the tab) some models are being displayed properly but the materials/textures tab is not working fine, may be it can be related to the scene-graph API or just for the editor.

I think on some material of the model, pbrMetallicRoughness is undefined and there is no way to catch it up. like this scenario.

I'm attaching screenshot of the errors what I'm getting in console.

![Screenshot from 2021-08-06 16-05-49](https://user-images.githubusercontent.com/30554980/128502466-c2f9b578-aa9c-4b84-8aec-997b82b90dbb.png)
 
![Screenshot from 2021-08-06 16-06-19](https://user-images.githubusercontent.com/30554980/128502492-9f695c95-57ba-46fc-8247-adb65b14b59c.png)

**Comment by elalish** [comment]
Can you please include the problem model(s) so I can repro?

**Reply by shamsail** [reply]
Yeah sure. sorry for the delay. I'm attaching the model link and a screenshot of the editor with error. So, that you can check.
Please find the model..

https://www.dropbox.com/s/3fw1kg08zjcdxu3/davinci_02.glb?dl=0
![Screenshot from 2021-08-09 16-32-31](https://user-images.githubusercontent.com/30554980/128700310-a29be6ac-7416-48ef-9957-f17f348f96df.png)

In the screenshot, you can see that there are no materials in the tab (because there is no catch in the editor to catch the null/undefined values.)(It should be implemented in the editor I guess or is it related to the upcoming version?)

**Comment by ghost** [comment]
Submitting a fix for this
I found that there are two materials 'default material' and 'skybox' that creating this issue. They don't seem to represented by Three.js and we are just creating an empty shell of a material.
Workaround:
I found that by re-exporting the model with gltf-pipeline these extraneous materials are removed and everything works fine.
you can find install instructions for gltf-pipeline here: https://github.com/CesiumGS/gltf-pipeline
I ran:
```
gltf-pipeline -i davinci_02.glb -o davinci_03.glb
```
That stripped it of these extra materials and then it was fine. Not sure where they are coming from, perhaps the tool your using to export is using a gltf-extension not handled by three.js?
